### ISSUE TITLE
Inline bold [B] label formatting causes some RTL Arabic text to be shaved on textbox edges

### ISSUE THREAD

**Issue by jurialmunkey** [xbmc/xbmc#22547]
### Describe the bug
Here is a clear and concise description of what the problem is:

When using `[B]` inline tags to bold format RTL  Arabic text, some text is occassionally shaved on the edges of the textbox. This issue did not previously occur in Matrix.

For example, compare the below two textboxes which are identical except that the one with magenta text uses `[B]` inline formatting whereas the limegreen text does not. On the second line (see yellow underline) you can see the character has been shaved off in the magenta textbox when compared to the limegreen version.
![Screenshot 2023-01-22 151912](https://user-images.githubusercontent.com/4618719/213900485-ac1bd2de-0c0a-4812-bda7-65716f90310b.jpg)

Depending on the exact text used and the textbox dimensions, the effect can be quite significant as seen in this issue thread on my skin: https://github.com/jurialmunkey/skin.arctic.horizon.2/issues/765

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Textboxes should adjust line returns appropriately to accomodate bold inline formatting

## Actual Behavior

Textboxes do not properly adjust to bold inline formatting and on occassion bolded text will fall outside the bounds.

## Possible Fix

~Not sure about a possible fix but I believe it is likely inadvertently related to the font texture atlas being more tightly packed in~ ~https://github.com/xbmc/xbmc/pull/21515~

~I suspect the issue has always been there but the tighter atlas packing makes it easier to reproduce~

EDIT: Tested with nightly (2022-05-30) prior to atlas packing PR and issue still exists.

### To Reproduce

Add some placeholder text as a variable in Variables.xml

```xml
<variable name="TestLabel"><value>لكن لا بد أن أوضح لك أن كل هذه الأفكار المغلوطة حول ا</value></variable>
```

Then in Home.xml (or anywhere), add some comparison textboxes:

```xml
<control type="textbox">
	<left>400</left>
	<width>490</width>
	<height>200</height>
	<label>$VAR[TestLabel,[COLOR=magenta][B],[/B][/COLOR] - ]$VAR[TestLabel]</label>
</control>
<control type="textbox">
	<left>1000</left>
	<width>490</width>
	<height>200</height>
	<label>$VAR[TestLabel,[COLOR=limegreen],[/COLOR] - ]$VAR[TestLabel]</label>
</control>
```

I'm using a 4k (3840x2160) resolution in Kodi, so I'm not sure if you'd need to adjust the textbox width slightly to reproduce on other resolutions. These are the dimensions I was able to reproduce with however.

Also note you'll need to switch to the "Arial" fontset in Estuary.

### Debuglog

The debuglog can be found here:

https://paste.kodi.tv/ahetigevob.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

See above.

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 11
 - Kodi version: Nexus (main release)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by sarbes** [comment]
> Not sure about a possible fix but I believe it is likely inadvertently related to the font texture atlas being more tightly packed in: https://github.com/xbmc/xbmc/pull/21515

I don't think so. From the provided screenshots, \u0627 doesn't seem to be part of a larger sequence and occurs multiple times accessing the same atlas entry. It is rendering fine in all other locations. The kerning looks fine.

IMO, it is unlikely that the PR mentioned has something to do with the issue, as it is only influencing the character position on the texture, but not the parameters of the glyph. Dumping the texture and overlaying the geometry via Renderdoc should bring some more insights.

If I had to guess, I'd say that the line is too wide and the geometry gets cropped off.

**Comment by psp2111-ADSLGATE** [comment]
Thanks @jurialmunkey for you insight on the bug, but I dont think making the font Bold is the issue. 
it is also happening with normal size font. 

at first I thought it is because the text is "Left align" instead of "Right align", but Kodi 19 also "Left align" with no issue.

I am with @sarbes I think the text line is too wide and goes beyond its container or window for some reason.

more pictures for the bug:

Estuary: 
Kodi 20
 
![Screenshot_58](https://user-images.githubusercontent.com/80945174/213915174-01f2b9f2-d6f8-4fe5-b75f-7477f283c17e.png)

Kodi 19

![Screenshot_59](https://user-images.githubusercontent.com/80945174/213915205-6a58bb75-1ee1-44ea-99a1-8c6d0c31ee03.png)

Arctic Horizon 2:
Kodi 20

![Screenshot_54](https://user-images.githubusercontent.com/80945174/213915247-c6f3b1c8-9f4c-46e9-8a49-02ce5e0f72e1.png)

Kodi 19

![Screenshot_56](https://user-images.githubusercontent.com/80945174/213915255-4f6671d9-0917-4d36-a6c6-2cef49d5ac35.png)

**Comment by jurialmunkey** [comment]
> 
> IMO, it is unlikely that the PR mentioned has something to do with the issue, as it is only influencing the character position on the texture, but not the parameters of the glyph. 

Ah, gotcha. To be clear, I wasn't suggesting the PR was the cause; I was only thinking less leeway on the texture edge potentially made it more likely to spill over if the underlying line width calculation was already off (i.e. by allowing it to push closer to the edge). If the atlas doesn't change glyph parameters at all then I can see how that wouldn't make any difference - simply me misunderstanding how it works.

To make absolutely certain, I tested with a nightly (2022-05-30) before that PR and could still reproduce. So definitely entirely unrelated and issue was introduced before that date.
